### ISSUE TITLE
Create enum constant - quick fix

### ISSUE THREAD

**Issue by FMorschel** [dart-lang/sdk#59679]
I'd like to request a quick-fix "Create enum constant {0}".

When you have code like:

```dart
enum E {
  one,
  two
}

void foo() {
  E.three^
}
```

There could be an assist here that would create an enum constant for this enum. I'm not sure about enums with constructors that require values if that is something we'd handle here, just to think about it.

I was thinking of it in a similar way to "Create method/function".

**Comment by bwilkerson** [comment]
I think it makes sense. I'd probably initially not support adding constants when the constructor has a non-zero number of arguments. If there's a good way to support the more general case we can always add it later.

**Comment by FMorschel** [comment]
Just to note here this should probably take https://github.com/dart-lang/sdk/issues/57036 into consideration. I've not looked if there has been anything here that would change the analysis, but there may be something.

**Comment by FMorschel** [comment]
While taking a look at this today, I found out that we already have an `Add enum constant '{0}'` fix. Not sure how I missed that (it seems to have been around for three years now).

I've noticed it never takes into consideration the assigning scope, which could be a problem, but since we now have enhanced enums with proprieties we can have a getter or something that returns something valid.

I'll close this for now and keep an eye out for why I missed it (I could not think of a case that doesn't show the fix on my current tests).`

If I ever do find out what happened for me to miss it (either missing `undefined_enum_constant` or a case where the fix is not working) I'll open a new issue about that specifically.
